src/tools/add_entity.cc

```diff
@@ -93,47 +101,28 @@ struct add_entity_tool : tool {
     }
 
     void alt_use(raycast_info *rc) override {
-        entity_name_index++;
-        if (entity_name_index>= entity_names.size()) {
-            entity_name_index = 0;
+        do {
+            entity_name_index++;
+            if (entity_name_index >= entity_names.size()) {
+                entity_name_index = 0;
+            }
+        } while (!entity_stubs[entity_names[entity_name_index]].get_component<placeable_component_stub>());
+
+        auto rotate = get_rotate();
+        if (rotate == 0) {
+            cur_rotate = 0;
+        }
+        else {
+            cur_rotate += rotate / 2;
+            cur_rotate = cur_rotate - (cur_rotate % rotate);
```

Comment:
Is this equivalent to the two lines: `cur_rotate -= (cur_rotate + rotate / 2) % rotate` ?

Reply:
probably

Reply:
uh, no, seems to not

Reply:
Assume the idea was to snap to the closest valid rotation in the new regime?

Reply:
I see, no, it's not quite the same. This is hard to follow because of cur_rotate appearing /everywhere/ in these expressions.

Reply:
yes. the problem here is shown when going from a rotated not to multiple of 90 entity to the door, which we want axis aligned. the door would rotate properly, but wouldn't be axis aligned.